### PR TITLE
Traffic Restrictions, Pass Build on CircleCI and local Windows 10 VS2019

### PR DESCRIPTION
**Pull request by joshuafc**
Traffic Restrictions, Pass Build on CircleCI (windows, ubuntu) and local Windows 10 VS2019

build info:  https://app.circleci.com/pipelines/github/joshuafc

Ubuntu, Windows Tested.

Some CI/CD configs added to repo from https://github.com/Toxe/example-cpp-ci-project, just delete them if not need.

### CODE REVIEW

**Comment by joshuafc** [comment]
travis take long time to wait queue， Maybe CircleCI could have a try. In my case, It start build immediately after push。

**Comment by skyformat99** [comment]
Very good.
With this  n2n can build  automatically.

File: src/edge_utils.c [hunk 1]
```diff
@@ -2055,8 +2055,8 @@ void readFromIPSocket(n2n_edge_t * eee, int in_sock) {
                     traceEvent(TRACE_NORMAL, "Supernode '%s' added to the list of supernodes.", sn->ip_addr);
                   }
 
-                  tmp_sock = (void*)tmp_sock + REG_SUPER_ACK_PAYLOAD_ENTRY_SIZE;
-                  tmp_mac = (void*)tmp_sock + sizeof(n2n_sock_t);
+                  tmp_sock = (char*)tmp_sock + REG_SUPER_ACK_PAYLOAD_ENTRY_SIZE;
```
Comment (skyformat99) [inline src/edge_utils.c]: ref  :  added payload struct to carry supernode data #501
The  #501 might be a better fix.

**Comment by joshuafc** [comment]
please, merge #501 first, after that, I will follow the change from main dev branch.

File: src/edge_utils.c (hunk 1 above)
Comment (joshuafc) [reply src/edge_utils.c]: please, merge #501 first, after that, I will follow the change from main dev branch.

**Comment by lucaderi** [comment]
@joshuafc I have merged #501, please update this PR for merging. Thanks

**Comment by joshuafc** [comment]
@lucaderi Codes on main branch cannot build on windows now.

I merged the code. But it cannot build on windows as main dev branch.

**Comment by joshuafc** [comment]
@lucaderi I have modify `n2n_win32.h` & `n2n_typedefs.h`.  Now, The PR's last commit pass build on windows and linux.

**Comment by lucaderi** [comment]
@joshuafc Your code is kinda broken as it still has partially overlapping datatypes. Please provide a PR to addess these issues

gcc -g  -I ./include        -c -o src/network_traffic_filter.o src/network_traffic_filter.c
src/network_traffic_filter.c:371:35: warning: incompatible pointer types assigning to 'n2n_verdict (*)(struct
      network_traffic_filter *, n2n_edge_t *, const n2n_sock_t *, uint8_t *, uint16_t)' (aka 'n2n_verdict (*)(struct
      network_traffic_filter *, struct n2n_edge *, const struct n2n_sock *, unsigned char *, unsigned short)') from 'n2n_verdict
      (network_traffic_filter_impl_t *, n2n_edge_t *, const n2n_sock_t *, uint8_t *, uint16_t)' (aka 'n2n_verdict (struct
      network_traffic_filter_impl *, struct n2n_edge *, const struct n2n_sock *, unsigned char *, unsigned short)')
      [-Wincompatible-pointer-types]
  filter->filter_packet_from_peer = filter_packet_from_peer;
                                  ^ ~~~~~~~~~~~~~~~~~~~~~~~
src/network_traffic_filter.c:372:34: warning: incompatible pointer types assigning to 'n2n_verdict (*)(struct
      network_traffic_filter *, n2n_edge_t *, uint8_t *, uint16_t)' (aka 'n2n_verdict (*)(struct network_traffic_filter *,
      struct n2n_edge *, unsigned char *, unsigned short)') from 'n2n_verdict (network_traffic_filter_impl_t *, n2n_edge_t *,
      uint8_t *, uint16_t)' (aka 'n2n_verdict (struct network_traffic_filter_impl *, struct n2n_edge *, unsigned char *,
      unsigned short)') [-Wincompatible-pointer-types]
  filter->filter_packet_from_tap = filter_packet_from_tap;
                                 ^ ~~~~~~~~~~~~~~~~~~~~~~
src/network_traffic_filter.c:373:10: warning: incompatible pointer types returning 'network_traffic_filter_impl_t *' (aka
      'struct network_traffic_filter_impl *') from a function with result type 'network_traffic_filter_t *'
      (aka 'struct network_traffic_filter *') [-Wincompatible-pointer-types]
  return filter;
         ^~~~~~
src/network_traffic_filter.c:377:34: warning: incompatible pointer types initializing 'network_traffic_filter_impl_t *' (aka
      'struct network_traffic_filter_impl *') with an expression of type 'network_traffic_filter_t *'
      (aka 'struct network_traffic_filter *') [-Wincompatible-pointer-types]
  network_traffic_filter_impl_t *_filter = filter;
                                 ^         ~~~~~~
src/network_traffic_filter.c:400:36: warning: incompatible pointer types initializing 'network_traffic_filter_impl_t *' (aka
      'struct network_traffic_filter_impl *') with an expression of type 'network_traffic_filter_t *'
      (aka 'struct network_traffic_filter *') [-Wincompatible-pointer-types]
    network_traffic_filter_impl_t *_filter = filter;
                                   ^         ~~~~~~

**Pull request merged**